### PR TITLE
fix(updates.jenkins.io): add `updates-jenkins-io` File Share

### PR DESCRIPTION
**Pull request by lemeurherve**
Follow-up of #445 

Ref: https://github.com/jenkins-infra/helpdesk/issues/2649#issuecomment-1660258021

### CODE REVIEW

File: updates.jenkins.io.tf [hunk 1]
```diff
@@ -17,6 +17,12 @@ resource "azurerm_storage_account" "updates_jenkins_io" {
   tags = local.default_tags
 }
 
+resource "azurerm_storage_share" "updates_jenkins_io" {
+  name                 = "updates-jenkins-io"
+  storage_account_name = azurerm_storage_account.updates_jenkins_io.name
+  quota                = 2 # Current updates.jenkins.io total size in /www/updates.jenkins.io: 400Mo
```
Comment (dduportal) [inline updates.jenkins.io.tf:23]: 2 Gb looks fine for this one. Note for later: could be nice to have this value synced to the PV/PVC quotas

Comment (dduportal) [inline updates.jenkins.io.tf:23]:
```suggestion
  quota                = 2 # Expected updates.jenkins.io total size in /www/updates.jenkins.io: 400Mo
```

Current deos not map to a specific timestamp ;)

Comment (lemeurherve) [reply updates.jenkins.io.tf:23]: WDYT of
```suggestion
  quota                = 2 # updates.jenkins.io total size in /www/updates.jenkins.io: 400Mo (Mid 2023)
```

**Review by dduportal — APPROVED** [review]
LGTM

**Comment by dduportal** [comment]
Note: this PR fixes a `Mount error(2): No such file or directory` error when an `azurefile` is mounted as AKS PV (ref. https://learn.microsoft.com/en-us/troubleshoot/azure/azure-kubernetes/fail-to-mount-azure-file-share#mounterror2)

**Pull request merged**